API_DOCUMENTATION.md

```diff
@@ -0,0 +1,414 @@
+- [Get contracts in store](#get-contracts-in-store)
+- [Subscribing to changes](#subscribing-to-changes)
+- [Error management](#error-management)
+
+## Installation
+
+Using NPM:
+
+```
+npm install nucleojs --save
+```
+
+Using Yarn:
+
+```
+yarn add nucleojs
+```
+
+## Usage
+
+```javascript
+import {
+  NucleoString,
+  NucleoNumber,
+  NucleoBoolean,
+  NucleoObject,
+  createStore
+} from 'nucleojs';
+
+const fullNameContract = new NucleoObject({
+  name: 'fullName',
+  fields:  {
+    firstName: NucleoString,
+    lastName: NucleoString
+  }
+});
+
+const userContract = new NucleoObject({
+  name: 'user',
+  fields: {
+    name: fullNameContract,
+    age: NucleoNumber,
+    disabled: NucleoBoolean
+  }
+});
+
+const productsContract = new NucleoObject({
+  name: 'products',
+  fields: {
+    title: NucleoString,
+    price: NucleoNumber
+  }
+});
+
+const contracts = {
+  user: userContract,
+  products: productsContract
+};
+
+const store = createStore(contracts);
+const { dispatch, update, getStore, subscribe } = store;
+
+const user = dispatch('user')({ name: { firstName: 'John', lastName: 'Nor' }, age: 27 });
+// it'll save the data to store properly
+
+console.log(user);
+/*
+{
+  status: 'OK',
+  errors: [],
+  data: {
+    name: {
+      firstName: 'John',
+      lastName: 'Nor'
+    },
+    age: 27
+  },
+}
+*/
+```
+
+## Contracts
+
+### Todo
+
+## Types
+
+### Creating NucleoObject
+
+```javascript
+NucleoObject({
+  name: String,
+  fields:  NucleoObject
+});
+```
+
+**Usage**
+
+```javascript
+import {
+  NucleoString,
+  NucleoNumber,
+  NucleoBoolean,
+  NucleoObject
+} from 'nucleojs'
+
+const fullNameContract = new NucleoObject({
+  name: 'fullName',
+  fields:  {
+    firstName: NucleoString,
+    lastName: NucleoString
+  }
+});
+
+const userContract = new NucleoObject({
+  name: 'user', // don't need to be the same name as the variable, but need to be unique
+  fields: {
+    name: fullNameContract,
+    age: NucleoNumber,
+    disabled: NucleoBoolean
+  }
+});
+
+const productsContract = new NucleoObject({
+  name: 'products',
+  fields: {
+    title: NucleoString,
+    price: NucleoNumber
+  }
+});
+
+const contracts = {
+  user: userContract,
+  products: productsContract
+};
+```
+
+### Creating NucleoList
+
+```javascript
+NucleoObject(NucleoObject);
+```
+
+**Usage**
+
+```javascript
+import {
+  NucleoString,
+  NucleoNumber,
+  NucleoBoolean,
+  NucleoObject,
+  NucleoList
+} from 'nucleojs'
+
+const fullNameContract = new NucleoObject({
+  name: 'fullName',
+  fields:  {
+    firstName: NucleoString,
+    lastName: NucleoString
+  }
+});
+
+const userContract = new NucleoObject({
+  name: 'user', // don't need to be the same name as the variable, but need to be unique
+  fields: {
+    name: fullNameContract,
+    age: NucleoNumber,
+    disabled: NucleoBoolean,
+  }
+});
+
+const usersContract = new NucleoList(userContract);
+
+const itemsContract = new NucleoList(NucleoString);
```

Comment:
In Nucleo, every contract in store must be a `NucleoObject`, so the usage for NucleoList would be closer to this:

```javascript
const userAccountsType = new NucleoObject({
  name: 'userAccountsType',
  fields: {
    accountType: NucleoString,
    accountNumber: NucleoNumber
  }
});

const personalInfoType = new NucleoObject({
  name: 'personalInfo',
  fields:  {
    firstName: NucleoString,
    lastName: NucleoString,
    items: new NucleoList(NucleoString),
    accounts: new NucleoList(userAccountsType)
  }
});

const userType = new NucleoObject({
  name: 'user',
  fields: {
    personalInfo: personalInfoType,
    age: NucleoNumber
  }
});
const contracts = { user: userType };
const store = createStore(contracts);
```

In this code, inside user has fields which are `NucleoLists`